main.py

```diff
@@ -105,11 +105,14 @@ def get_next_location(distances_df, km_left):
         apply_distance_calc, args=(start_lat, start_lon), axis=1
     )
     distances_df["ratio"] = distances_df.apply(calculate_ratio, axis=1)
-    best_next_location = pd.Series(distances_df.query("distance < 1000").ratio.idxmax())
-    if distances_df.distance_to_home.loc[best_next_location].values[0] > km_left:
+    best_next_location = distances_df.query("distance < 1000").ratio.idxmax()
+    if distances_df.loc[best_next_location, "distance_to_home"] > km_left:
         return pd.Series(dtype="float64")
     else:
-        return distances_df.loc[best_next_location]
+        distances_df = distances_df.rename_axis("brewery_id").reset_index()
```

Comment:
Some dark magic is happening here.

Reply:
When creating pandas Series, Index(In this case, brewery_id) is taken as Series name. Because brewery_id is important, I move it to a new column and reindex, so that brewery_id is not lost in Series name.